python/lsst/pipe/tasks/skyCorrection.py

```diff
@@ -339,67 +345,194 @@ def run(self, calExps, calBkgs, skyFrames, camera, backgroundToPhotometricRatioH
+                    warnings.filterwarnings("ignore", r"invalid value encountered")
+                    calBkgElement = bgModel2.toCcdBackground(detector, detector.getBBox())
+                skyCorr.append(calBkgElement[0])
+
+        # Make camera-level mosaics of bg subtracted calexps and subtracted bgs
+        calExpsBinned = []
+        calBkgsBinned = []
+        for calExpHandle, mask, skyCorr in zip(calExps, masks, skyCorrs):
+            calExp = self._getCalExp(calExpHandle, mask, skyCorr, backgroundToPhotometricRatioHandle)
+
+            skyCorrExtra = skyCorr.clone()  # new skyCorr elements created in this task
+            skyCorrExtra._backgrounds = skyCorrExtra._backgrounds[bgModel1Index:]
+            skyCorrExtraMI = makeMaskedImage(skyCorrExtra.getImage())
+            skyCorrExtraMI.setMask(calExp.getMask())
+
+            calExpsBinned.append(binImage(calExp.getMaskedImage(), self.config.binning))
+            calBkgsBinned.append(binImage(skyCorrExtraMI, self.config.binning))
+
+        mosConfig = VisualizeMosaicExpConfig()
+        mosConfig.binning = self.config.binning
+        mosTask = VisualizeMosaicExpTask(config=mosConfig)
+        detectorIds = [detector.getId() for detector in detectors]
+        calExpMosaic = mosTask.run(calExpsBinned, camera, inputIds=detectorIds).outputData
+        calBkgMosaic = mosTask.run(calBkgsBinned, camera, inputIds=detectorIds).outputData
 
         return Struct(
-            skyFrameScale=skyFrameScale, skyCorr=calBkgs, calExpMosaic=calExpMosaic, calBkgMosaic=calBkgMosaic
+            skyFrameScale=skyFrameScale,
+            skyCorr=skyCorrs,
+            calExpMosaic=calExpMosaic,
+            calBkgMosaic=calBkgMosaic,
         )
 
-    def _restoreOriginalBackgroundRefineMask(self, calExps, calBkgs):
-        """Restore original background to each calexp and invert the related
+    def _getCalExp(self, calExpHandle, mask=None, skyCorr=None, backgroundToPhotometricRatioHandle=None):
+        """Get a calexp from a DeferredDatasetHandle, and optionally apply an
+        updated mask and skyCorr.
+
+        Parameters
+        ----------
+        calExpHandle : `~lsst.afw.image.ExposureF`
+                | `lsst.daf.butler.DeferredDatasetHandle`
+            Either the image exposure data or a handle to the calexp dataset.
+        mask : `lsst.afw.image.Mask`, optional
+            Mask to apply to the calexp.
+        skyCorr : `lsst.afw.math.BackgroundList`, optional
+            Background list to subtract from the calexp.
+
+        Returns
+        -------
+        calExp : `lsst.afw.image.ExposureF`
+            The calexp with the mask and skyCorr applied.
+        """
+        if isinstance(calExpHandle, DeferredDatasetHandle):
+            calExp: ExposureF = calExpHandle.get()
+        else:
+            # Here we clone the imaging data to avoid modifying data which is
+            # used in downstream processing.
+            calExp: ExposureF = calExpHandle.clone()
```

Comment:
I don't see `from __future__ import annotations`. Does this type hint work without that?

Reply:
Thank you! Added.